### PR TITLE
Prevent tab completion if the word to complete is a single dash

### LINKED ISSUE #16577: functions with `-` on the name wrongfully selected for tab completion
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://www.powershellgallery.com/packages/PSReadLine)
- [X] Search the existing issues, especially the pinned issues.

### Exception report

```console
n/a
```

### Screenshot

image

### Environment data

```console
PS Version: 7.2.0
PS HostName: ConsoleHost (Windows Terminal)
PSReadLine Version: 2.2.0-beta4
PSReadLine EditMode: Windows
OS: 10.0.19041.1288 (WinBuild.160101.0800)
BufferWidth: 120
BufferHeight: 35
```

### Steps to reproduce

1. in a clean pwsh declare any function with a `-` in it's name, i used:
```poweshell
function test-{}
function test-2{}
```
2. type any command (preferably one that doesn't have any completer) followed by a `-`
3. press `tab` or `ctrl+space`

### Expected behavior

nothing?

### Actual behavior

the functions with `-` in the name gets inserted

### PR DESCRIPTION
**Pull request by ayousuf23**
# PR Summary

When using tab completion, functions with `-` are wrongfully selected. In order to prevent this, an additional check is added in this PR to prevent tab completion when the word to complete is a single dash.

## PR Context

This PR resolves #16577. It simply adds an additional condition (making sure the word to complete is not a single dash) that should be satisfied before PowerShell performs tab completion.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [ ] N/A or can only be tested interactively
    - **OR**
    - [x] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and opened an issue in the relevant tool repository. This may include:
        - [ ] Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode
        (which runs in a different PS Host).
            - [ ] Issue filed: 
        - [ ] Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
            - [ ] Issue filed: 
        - [ ] Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
            - [ ] Issue filed: 
        - [ ] Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).
            - [ ] Issue filed:

### CODE REVIEW

File: src/System.Management.Automation/engine/CommandCompletion/CompletionCompleters.cs [hunk 1]
```diff
@@ -1329,7 +1329,7 @@ internal static List<CompletionResult> CompleteCommandArgument(CompletionContext
                         context.Options.Remove("LiteralPaths");
                 }
 
-                if (context.WordToComplete != string.Empty && context.WordToComplete.Contains('-'))
+                if (context.WordToComplete != string.Empty && context.WordToComplete != "-" && context.WordToComplete.Contains('-'))
```
Comment (iSazonov) [inline src/System.Management.Automation/engine/CommandCompletion/CompletionCompleters.cs:1332]:
```suggestion
                if (context.WordToComplete != string.Empty && !context.WordToComplete.StartsWith('-') && context.WordToComplete.Contains('-'))
```

**Comment by ayousuf23** [comment]
@iSazonov I can't see your suggestion/code review anymore. Can you please add it again?

**Comment by ayousuf23** [comment]
@iSazonov Nevermind. I found your suggestion.

**Comment by iSazonov** [comment]
> Nevermind. I found your suggestion and updated the code.

@ayousuf23 Sorry, I removed the post soon because it was wrong. Please remove your last commit and add a test for scenario you fix.

**Comment by ayousuf23** [comment]
@iSazonov I reverted the most recent commit. Where should I add the test?

**Comment by iSazonov** [comment]
> @iSazonov I reverted the most recent commit. Where should I add the test?

I think in TabCompletion.Tests.ps1

**Comment by ayousuf23** [comment]
@iSazonov I added a test.

**Review by iSazonov — APPROVED** [review]

**Comment by ayousuf23** [comment]
@iSazonov Is there anything remaining for me to do in order to get the PR merged?

**Comment by iSazonov** [comment]
@ayousuf23 We are waiting @daxian-dbw review.

**Comment by ayousuf23** [comment]
@daxian-dbw Can you please review this PR?

**Review by daxian-dbw — APPROVED** [review]
@ayousuf23 Sorry for the delay of review.
Thanks for the contribution! I made a minor update and added a comment.

**Pull request merged**